packages/plugin-legacy/src/index.ts

```diff
@@ -331,24 +331,30 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
         }
 
         return (chunkInfo) => {
-          let fileName =
+          const file =
             typeof fileNames === 'function' ? fileNames(chunkInfo) : fileNames
+          const fileArr = file.split('/')
+          let fileName = fileArr.pop()!
```

Comment:
Reading rollup docs, they only mention forward slashes, so I think this should be good and a user passing windows style paths here is not supported https://rollupjs.org/configuration-options/#output-assetfilenames

Reply:
What is the reason to only apply the conversion to the file part? I guess `/foo/[name]/[hash].[ext]` is valid.

Reply:
Oh... I totally overlooked that. Thanks for reviewing this sapphi. I guess we need to go with something like I had at https://github.com/vitejs/vite/pull/15115#issuecomment-1824073000

Reply:
> What is the reason to only apply the conversion to the file part? I guess /foo/[name]/[hash].[ext] is valid.

Although rollup allows this use, there is no mention of this usage in all related documentation and the default values ​​of the properties, and these placeholders: [name], [hash], [ext], [extname] are strictly speaking only can be used for the final file name.